### ISSUE TITLE
Bound storage costs of `CoalescenceTimeDistribution` by pre-specifying time windows

### ISSUE THREAD

**Issue by nspope** [tskit-dev/tskit#2613]
#2119 implemented a class `CoalescenceTimeDistribution` that internally builds a table where each row is a node and columns are combinatorial weights. This allows the weighted ECDF and coalescence rates to be calculated very quickly for arbitrary time points without re-traversing the tree sequence, but the storage costs can get very large for tree sequences with many samples (or when there are many types of weights). So currently the API looks like:
```python
coaltime = ts.coalescence_time_distribution(
    sample_sets,
    weight_func='pair_coalescence_events',
    window_breaks=genomic_windows,
) #builds table
coaltime.coalescence_rate_in_intervals(time_windows) #very fast
```

An alternative would be to sum weights across nodes within time windows. In this case, the ECDF can't be evaluated at arbitrary time points, only at the breakpoints of the time windows. However, the storage costs are bounded. Then, the API would look something like,
```python
coaltime = ts.coalescence_time_distribution(
    sample_sets,
    weight_func='pair_coalescence_events',
    window_breaks=genomic_windows,
    time_breaks=time_windows,
) 
coaltime.coalescence_rate_in_intervals()
```

Because the machinery is similar between these two cases (aside from a summation operation), I think it'd be possible to keep the original behavior when `ts.coalescence_time_distribution(..., time_breaks=None)`.

**Comment by nspope** [comment]
Nope-- this (and other issues regarding `CoalescenceTimeDistribution`) are no longer relevant.

**Comment by hyanwong** [comment]
Can remove the "future" tag then?
